app/models/name.rb

```diff
@@ -1639,22 +1679,33 @@ def self.parse_author(str)
   end
 
   def self.parse_group(str, deprecated = false)
-    results = nil
-    if match = GROUP_PAT.match(str)
-      name = match[1]
-      text_name = name.tr("ë", "e")
-      parent_name = name.sub(LAST_PART, "")
-      results = ParsedName.new(
-        text_name: text_name + " group",
-        search_name: text_name + " group",
-        sort_name: format_sort_name(text_name, "group"),
-        display_name: format_name(name, deprecated) + " group",
-        parent_name: parent_name,
-        rank: :Group,
-        author: ""
-      )
+    return unless match = GROUP_PAT.match(str)
+
+    # Parse the string without "group"
+    ungrouped_str = str.sub(/\s#{GROUP_ABBR}/, "")
+    result = parse_name(ungrouped_str)
+    return nil unless result
+
+    # Adjust the parsed name
+    result.text_name += " group"
+
+    if result.author.present?
+      # Add "group" before author
+      author = Regexp.escape(result.author)
+      result.search_name.sub!( /(.*)(#{author})/, '\1group \2')
+      result.sort_name.sub!(   /(.*)(#{author})/, '\1 group  \2')
+      result.display_name.sub!(/(.*)(#{author})/, '\1group \2')
+    else
```

Comment:
As I was getting ready to replace
`result.search_name.sub!( /(.*)(#{author})/, '\1group \2')`, etc., 
with
`result.search_name.sub!(/(#{author})$/, 'group \1')`, etc.,
I finally realized -- given our assumption that author is at the end -- that the two `sub`s have the same functionality. The `*` is greedy, so `(.*)` gobbles up everything until the **last** occurrence of `author`.
Your suggestion is simpler (only 1 capture group), shorter, and clearer.
The other way is marginally more flexible -- it would still work if we ever decided to allow something (I have no clue what) to appear after author. Maybe that's just premature generalization.
**What do you think?**